src/OpenTelemetry.Exporter.Zipkin/Implementation/TraceExporterHandler.cs

```diff
@@ -54,8 +56,26 @@ public async Task ExportAsync(IEnumerable<SpanData> spanDataList)
 
             foreach (var data in spanDataList)
             {
-                var zipkinSpan = this.GenerateSpan(data, this.localEndpoint);
-                zipkinSpans.Add(zipkinSpan);
+                bool shouldExport = true;
+                foreach (var label in data.Attributes.AttributeMap)
+                {
+                    if (label.Key == "http.url")
+                    {
+                        if (label.Value is string urlStr && urlStr == this.serviceEndpoint)
+                        {
+                            // do not track calls to Zipkin
+                            shouldExport = false;
+                        }
+
+                        break;
+                    }
+                }
+
+                if (shouldExport)
+                {
+                    var zipkinSpan = this.GenerateSpan(data, this.localEndpoint);
+                    zipkinSpans.Add(zipkinSpan);
```

Comment:
@lmolkova I might be reading the code incorrectly, but there appears to be a possibility that `SendSpansAsync` can be called with an empty span list, resulting in a HTTP call to Zipkin, which still causes a invisible loop.

Reply:
@johnduhart sorry, this is merged already... 

Would you mind sending very short PR to return from `ExportAsync` immediately if Span batch is empty? 
It seems it should not ever happen (but seems to be reasonable anyway), see here: https://github.com/open-telemetry/opentelemetry-dotnet/blob/8aae2c02f1760668734d99e848ca780e7635b578/src/OpenTelemetry/Trace/Export/SpanExporterWorker.cs#L99

And also after loop runs we should make sure zipkinSpans are not empty.

Reply:
> And also after loop runs we should make sure zipkinSpans are not empty.

This was my main concern, given we're selectively including spans in this method.

Sure, I'll cut a PR to add that.